### PR TITLE
[WIP] Enable `linalg.batch_matmul`

### PR DESCRIPTION
**Pull request by Abhishek-Varma**
This is a WIP PR.

As of now it contains 3 commits - (we should just be reviewing the 2nd one) :-
Commit 1 - @yzhang93 's fix patch for enabling lower sized `linalg.matmul`.

Commit 2 - The initial infra for `linalg.batch_matmul` <---- this changes the current matmul packing config too, so this will be addressed as part of this WIP PR + some comment cleaning up.

Commit 3 - @erwei-xilinx 's fix patch for `--iree-amdaie-decompose-pack-unpack-to-air` 3D -> 2D.

### CODE REVIEW

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp [hunk 1]
```diff
@@ -74,6 +117,8 @@ static LogicalResult setRootConfigImpl(func::FuncOp entryPointFn, Operation *op,
         // TODO (nmeshram): This is very limited for now, plan is to
         // let it first crash for all the other ops and then consiously
         // add support for them, this way we can verify our work.
+        .Case<linalg::ContractionOpInterface>(
+            [&](auto op) { return setRootConfig(entryPointFn, op); })
```
[the remark this answers is not in the document]
Comment (Abhishek-Varma) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp:121]: Yes, you're correct - just that I was trying to group both of these under the umbrella of `ContractionOpInterface`.
Haven't shifted linalg.matmul there - I'll address your point though.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEPackAndTranspose.cpp [hunk 2]
```diff
@@ -61,10 +64,11 @@ static FailureOr<PackConfig> getPackConfig(
     // 0 is used when unpack is empty
     config.unpackEmpty = {0};
     config.innerPerm = {{1, 0}};
-    config.outerPerm = {{0, 1}};
+    config.outerPerm = {{0, 1, 2}};
```
Comment (nirvedhmeshram) [inline compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEPackAndTranspose.cpp:67]: I believe this path is run for both matmul and batch matmul? If yes, I am wondering why it is okay add an extra permutation for the matmul. Perhaps the reason is that {0,1,2} is identity or no permutation at all? in which case we can drop this line all together?

**Review by nirvedhmeshram — COMMENTED** [review]
Thanks for this @Abhishek-Varma , currently what are the errors that we are facing e2e while running this? Can you link the related issues here so that we can track it in once place. it might make sense to start a issue to track all the batch mamtul issue that then link the sub issues there.

File: compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEPackAndTranspose.cpp (hunk 2 above)
Comment (Abhishek-Varma) [reply compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEPackAndTranspose.cpp:67]: Oh no no - I am not sure if that claim would be right. So, I'll update this logic based on the `numLoops`/`numDims`/etc.

Currently just added the infra that just targets the batch_matmul - working on cleaning it up - should be complete by tomorrow.

We can currently review of the tile sizes and the packing config I'm using particularly for batch_matmul case.

**Pull request closed without merging**